iceoryx_posh/include/iceoryx_posh/popo/modern_api/base_subscriber.hpp

```diff
@@ -98,10 +107,28 @@ class BaseSubscriber : public Condition
     ///
     void releaseQueuedSamples() noexcept;
 
-    // Condition overrides
-    virtual void setConditionVariable(ConditionVariableData* const conditionVariableDataPtr) noexcept override;
-    virtual void unsetConditionVariable() noexcept override;
-    virtual bool hasTriggered() const noexcept override;
+    /// @brief attaches a WaitSet to the subscriber
+    /// @param[in] waitset reference to the waitset to which the subscriber should be attached to
+    /// @param[in] subscriberEvent the event which should be attached
```

Comment:
events and triggerId need more explanation regarding the underlying assumptions and how they are supposed to be used. I guess there is a readme somewhere but maybe a short explanation here (as note?) would be useful as well.

Reply:
The `SubscriberEvent` is an enum which currently can be only set to `SubscriberEvent::HAS_NEW_SAMPLES`. If we add more events it is still completely readable and I think understandable from the context.

For the triggerId I added a more detailled documentation.